terminus/geometry/arc.py

```diff
@@ -34,11 +34,26 @@ def __init__(self, start_point, theta, radius, angular_length):
         is considered counter-clockwise
         """
         self._start_point = start_point
-        self._theta = theta
+        self._theta = theta % 360
         self._radius = radius
         self._angular_length = angular_length
         self._end_point = self._compute_point_at(angular_length)
 
+    @classmethod
+    def from_points_in_circle(cls, start_point, end_point, circle):
+        """
+        Returns a counter-clockwise arc from start point to end point on the given circle
+        """
+        center = circle.center
+        start_point_angle = Point(1, 0).angle((start_point - center))
+        start_heading = start_point_angle + 90
+        angle_between_vectors = (start_point - center).angle(end_point - center)
+        if angle_between_vectors >= 0:
+            angular_length = angle_between_vectors
+        else:
+            angular_length = 360 + angle_between_vectors
+        return Arc(start_point, start_heading, circle.radius, angular_length)
```

Comment:
You should use `cls` here (which in this case will be `Arc`, but allows you to later subclass it without problems). So something like `cls(...)`